code_generator/GdkCodeGenerator/Templates/UnityCommandPayloadGenerator.tt

```diff
@@ -18,14 +18,17 @@ namespace <#= qualifiedNamespace #>
 <# foreach (var commandDetails in commandDetailsList) {#>
         public class <#= commandDetails.CommandName #>
         {
+            /// <summary>
+            ///     Please do not use the default constructor. Use CreateRequest instead.
+            /// </summary>
             public struct Request
             {
                 public EntityId TargetEntityId { get; }
                 public <#= commandDetails.RequestType #> Payload { get; }
                 public uint? TimeoutMillis { get; }
                 public bool AllowShortCircuiting { get; }
 
-                public Request(EntityId targetEntityId,
+                internal Request(EntityId targetEntityId,
```

Comment:
If we are expecting users to create things via the static functions then we can inline this constructor in there.

Reply:
Yes - will need to make the properties `internal set` though